src/repositories/cart.repository.ts

```diff
@@ -1,37 +1,32 @@
-import { carts } from "../storage";
-import crypto from "crypto";
-import { Cart, Product } from "../types/types";
+import { Cart } from '../types/types';
+import { CartModel, convert } from '../models/cart.model';
+import { HttpError } from '../common/errors';
 
-export const getCart = (userId: string): Cart => {
-  let cart = carts.find((cart) => cart.userId === userId);
-
-  if (!cart) {
-    cart = {
-      id: crypto.randomUUID(),
-      userId,
+export const getCart = async (userId: string): Promise<Cart> => {
+  let cart = await CartModel.findOne({ user: userId }).populate(['user', 'products']);
+  if (cart == null) {

```

Comment:
This condition should be on the service level. The function for creating a new cart should be on the repository level